test/functional/ui/bufhl_spec.lua

```diff
@@ -275,4 +278,140 @@ describe('Buffer highlighting', function()
                                               |
     ]])
   end)
+
+  it('supports virtual text annotations', function()
+    local set_text = curbufmeths.set_virtual_text
```

Comment:
We should avoid much "aliasing" in the tests. Already `nvim_set_virtual_text` is abbreviated to `set_virtual_text`; then further abbreviating it to `set_text` makes the tests hard to read...